src/api/app/services/action_build_results_service/chart_data_extractor.rb

```diff
@@ -6,16 +6,24 @@ def initialize(actions:)
       @actions = actions
     end
 
+    # rubocop:disable Metrics/CyclomaticComplexity
     def call
       return [] if @actions.blank?
 
       @actions.where(type: %i[submit maintenance_incident maintenance_release]).map do |action|
         sources = sources_from_action(action)
         next unless sources[:source_project].present? && sources[:source_package].present?
 
-        package_build_results(sources[:source_package], sources[:source_project])
+        build_results = package_build_results(sources[:source_package], sources[:source_project])
+        targets = target_from_action(action) if BsRequest.find(action.bs_request_id).staging_project_id.nil?
+        unless targets.nil?
```

Comment:
If I'm not wrong you can shortcut this loop if `targets.nil?` so:
```
next build_results if targets.nil?
```
Would skip lines below, then you wouldn't need the `unless targets.nil?`

Reply:
Done. It has been updated